tools/reference_checker.cpp

```diff
@@ -201,17 +201,25 @@ int run(const ReferenceCheckerOptions &options) {
                      Flay::optimizeProgram(options.getCompilerOptions(), options.toFlayOptions()),
                      EXIT_FAILURE);
     std::stringstream flayOptimizationOutput;
-    for (const auto *node : flayServiceStatistics.eliminatedNodes) {
+    for (const auto &elimRepl : flayServiceStatistics.eliminatedNodes) {
+        auto [node, replaced] = elimRepl;
         if (node->getSourceInfo().isValid()) {
             auto sourceFragment = node->getSourceInfo().toSourceFragment(false).trim();
-            flayOptimizationOutput << "Eliminated node at line "
-                                   << node->getSourceInfo().toPosition().sourceLine << ": "
-                                   << sourceFragment;
+            if (replaced == nullptr) {
+                flayOptimizationOutput << "Eliminated node at line "
+                                       << node->getSourceInfo().toPosition().sourceLine << ": "
+                                       << sourceFragment;
+            } else {
+                // FIXME: any elegant solution to just print first line of `replaced->toString()`?
```

Comment:
Do not use `toString` it only gives incomplete node  information, use the source fragment. I think it is fine to have multiple lines. But we need to check how that looks like. Let's fix that in a separate PR. 

Also make sure the source info is valid. It should be by necessity of construction, but it is good to check that this invariant holds.

Reply:
Thanks. Will do that.